src/r_tipping/RateSystem.jl

```diff
@@ -0,0 +1,98 @@
+# we consider the ODE dxₜ/dt = f(xₜ,λ(rt))
+# here λ = λ(t) ∈ Rᵐ is a function containing all the system parameters 
+
+# We ask the user to define: 
+#  1) a ContinuousTimeDynamicalSystem that should be investigated and
+#  2) a protocol for the time-dependent forcing with the struct RateProtocol
+
+# Then we give back the ContinuousTimeDynamicalSystem with the parameter 
+# changing according to the rate protocol
+"""
+    RateProtocol
+
+The RateProtocol contains all the fields (information) that allows to take an ODE of the form 
+    `dxₜ/dt = f(xₜ, λ)` 
+with `λ ∈ Rᵐ` containing all system parameters, and make it an ODE of the form 
+    `dxₜ/dt = f(xₜ, λ(rt))`
+with `λ(t) ∈ Rᵐ` constant before time `t_start` and also after time `t_end`. 
+
+RateProtocol contains the following fields:
```

Comment:
The user shouldn't care about what fields the data structure has, which also allows you to attribute more private fields to it. Here you should just make a section called `## Keyword arguments` in harmony with the rest of the ecosystem.

Also, refer to the data structure in code since it is a code name (type). Use backticks and write `RateProtocol` instead of RateProtocol.

Reply:
Also, no need to declare the fields of the arguments.